sdk/videoanalyzer/video-analyzer-edge/README.md

```diff
@@ -0,0 +1,159 @@
+- To interact with Azure IoT Hub you will need to run `npm install azure-iothub`
+- You will need to use the version of the SDK that corresponds to the version of the Video Analyzer Edge module you are using.
+
+  | SDK     | Video Analyzer Edge Module |
+  | ------- | -------------------------- |
+  | 1.0.0b1 | 1.0                        |
+
+### Creating a pipeline topology and making requests
+
+Please visit the [Examples](#examples) for starter code.
+
+## Key concepts
+
+### Pipeline Topology vs Pipeline Instance
+
+A _pipeline topology_ is a blueprint or template for instantiating live pipelines. It defines the parameters of the pipeline using placeholders as values for them. A _live pipeline_ references a pipeline topology and specifies the parameters. This way you are able to have multiple live pipelines referencing the same topology but with different values for parameters. For more information please visit [pipeline topologies and live pipelines][doc_pipelines].
+
+## Examples
+
+### Creating a pipeline topology
+
+To create a pipeline topology you need to define parameters, sources, and sinks.
+
+```typescript
+const rtspSource: RtspSource = {
+    name: "rtspSource",
+    endpoint: {
+      url: "${rtspUrl}",
+      "@type": "#Microsoft.VideoAnalyzer.UnsecuredEndpoint",
+      credentials: {
+        username: "${rtspUserName}",
+        password: "${rtspPassword}",
+        "@type": "#Microsoft.VideoAnalyzer.UsernamePasswordCredentials"
+      }
+    } as UnsecuredEndpoint,
+    "@type": "#Microsoft.VideoAnalyzer.RtspSource"
+  };
+
+  const nodeInput: NodeInput = {
+    nodeName: "rtspSource"
+  };
+
+  const msgSink: IotHubMessageSink = {
+    name: "msgSink",
+    inputs: [nodeInput],
+    hubOutputName: "${hubSinkOutputName}",
+    "@type": "#Microsoft.VideoAnalyzer.IotHubMessageSink"
+  };
+
+  const pipelineTopology: PipelineTopology = {
+    name: "jsTestTopology",
+    properties: {
+      description: "description for jsTestTopology",
+      parameters: [
+        { name: "rtspUserName", type: "String", default: "dummyUsername" },
+        { name: "rtspPassword", type: "SecretString", default: "dumyPassword" },
+        { name: "rtspUrl", type: "String" }
+        { name: "hubSinkOutputName", type: "String" }
+      ],
+      sources: [rtspSource],
+      sinks: [msgSink]
+    }
+  };
+
+```
+
+### Creating a live pipeline
+
+To create a live pipeline instance, you need to have an existing pipeline topology.
+
+```typescript
+const livePipeline: LivePipeline = {
+  name: pipelineTopologyName,
+  properties: {
+    description: "description for jsTestLivePipeline",
+    topologyName: "jsTestTopology",
+    parameters: [{ name: "rtspUrl", value: "rtsp://sample.com" }]
+  }
+};
+```
+
+### Invoking a pipeline method request
+
+```typescript
+const pipelineTopologySetRequest = createRequest("pipelineTopologySet", pipelineTopology);
+const setPipelineTopResponse = await iotHubClient.invokeDeviceMethod(device_id, module_id, {
```

Comment:
where do they get device_id and module_id from? also these should be `deviceId` and `moduleId` probably

Reply:
They get that from their Azure Iot Hub. I added a sentence explaining that